orange/src/main/java/com/hartwig/hmftools/orange/algo/util/GermlineConversion.java

```diff
@@ -147,6 +161,96 @@ private static List<PurpleGainLoss> toSomaticGainsLosses(@Nullable List<PurpleGa
         return reportableGermlineGainsLosses != null ? reportableGermlineGainsLosses : Lists.newArrayList();
     }
 
+    @NotNull
+    private static List<PurpleGeneCopyNumber> getAdditionalSuspectGeneCopyNumberWithLOH(@NotNull PurpleRecord purple)
+    {
+        List<PurpleHeterozygousDeletion> reportableGermlineHeterozygousDeletions = purple.reportableGermlineHeterozygousDeletions();
+        if(reportableGermlineHeterozygousDeletions == null)
+        {
+            return Lists.newArrayList();
+        }
+
+        List<String> relevantGenes = reportableGermlineHeterozygousDeletions.stream()
+                .map(PurpleHeterozygousDeletion::gene)
+                .distinct()
+                .collect(Collectors.toList());
+
+        List<PurpleGeneCopyNumber> additionalSuspectGeneCopyNumberWithLOH = Lists.newArrayList();
+        for(String gene : relevantGenes)
+        {
+            PurpleGeneCopyNumber candidate = findMatchingGeneCopyNumber(gene, purple.allSomaticGeneCopyNumbers());
+
+            if(candidate != null
```

Comment:
Did you consider throwing an exception in case `candidate == null`? I can't think of a reason why that would be allowed to happen?

Reply:
Done